internal/rest/http_error.go

```diff
@@ -18,25 +18,33 @@ package rest
 
 import (
 	"fmt"
-	"io/ioutil"
+	"io"
 	"net/http"
 )
 
 type Error struct {
-	Text string
-	Code int
+	text string
+	code int
 }
 
 func NewError(code int, text string) *Error {
 	return &Error{
-		Code: code,
-		Text: text,
+		code: code,
+		text: text,
 	}
 }
 
+func (e Error) Text() string {
```

Comment:
Why did you encapsulate fields that they don't have any logic inside?

Reply:
In order to make it implement `errors.HTTPError` of CLC.